### PR TITLE
add ubuntu 16.04 support

### PR DESCRIPTION
**Pull request by bastelfreak**
This was mostly working, it just wasn't tested. I added the tests and fixed some edge cases.

### CODE REVIEW

File: manifests/dev.pp [hunk 1]
```diff
@@ -23,21 +24,31 @@
...
     ensure_packages(["${php::package_prefix}xml"], {
       ensure  => present,
-      require => Class['::apt::update'],
+      require => $require,
     })
 
     package { $real_package:
       ensure  => $ensure,
-      require => Class['::php::packages'],
+      require => $require,
```
Comment (alexjfisher) [inline manifests/dev.pp]: Are you sure this is correct?
